### PR TITLE
address more errorprone warnings

### PR DESCRIPTION
**Pull request by lsf37**

### CODE REVIEW

File: jflex/src/main/java/jflex/core/AbstractLexScan.java [hunk 1]
```diff
@@ -21,7 +21,9 @@
   int bufferSize = 16384;
 
   File file;
-  private Stack<File> files = new Stack<>();
+
+  @SuppressWarnings("JdkObsolete")
+  private final Stack<File> files = new Stack<>();
```
Comment (regisd) [inline jflex/src/main/java/jflex/core/AbstractLexScan.java]: I'm pretty sure `ArrayDeque` could offer the same features.

File: jflex/src/main/java/jflex/gui/GridPanel.java [hunk 2]
```diff
@@ -20,16 +20,15 @@
...
+  private final int rows;
 
-  private int hgap;
-  private int vgap;
+  private final int hgap;
+  private final int vgap;
 
-  private Vector<GridPanelConstraint> constraints = new Vector<>();
+  private final Vector<GridPanelConstraint> constraints = new Vector<>();
   private Insets insets = new Insets(0, 0, 0, 0);
```
Comment (regisd) [inline jflex/src/main/java/jflex/gui/GridPanel.java]: Consider using `ArrayList`

Comment (lsf37) [reply jflex/src/main/java/jflex/gui/GridPanel.java]: Yup, should. That's a separate warning, though ;-)

File: jflex/src/main/java/jflex/core/AbstractLexScan.java (hunk 1 above)
Comment (lsf37) [reply jflex/src/main/java/jflex/core/AbstractLexScan.java]: Happy in principle to change out Stack for ArrayDeque.

**Comment by lsf37** [comment]
I have another set coming up that addresses both of these.

**Pull request merged**